### PR TITLE
Entry points ikz plugin

### PR DESCRIPTION
**Pull request by aalbino2**
@hampusnasstrom finally, they should be all there

### CODE REVIEW

File: IKZ_plugin/src/ikz_plugin/general/__init__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    def load(self):
+        from ikz_plugin.general.schema import m_package
+
+        return m_package
+
+
+general_schema = GeneralEntryPoint(
+    name='GeneralSchema',
+    description='Schema package defined using the new plugin mechanism.',
```
Comment (hampusnasstrom) [inline IKZ_plugin/src/ikz_plugin/general/__init__.py:16]: Add description:
```suggestion
    description='Schema package for general definitions used throughout IKZ.',
```

File: IKZ_plugin/src/ikz_plugin/movpe/movpe1/growth_excel/parser.py [hunk 2]
```diff
@@ -119,15 +117,9 @@ class RawFileMovpeDepositionControl(EntryData):
...
-    def __init__(self):
-        super().__init__(
-            name='MOVPE 1 Deposition Control IKZ',
-            code_name='MOVPE 1 Deposition Control IKZ',
-            code_homepage='https://github.com/FAIRmat-NFDI/AreaA-data_modeling_and_schemas',
-            supported_compressions=['gz', 'bz2', 'xz'],
-        )
-
     def parse(self, mainfile: str, archive: EntryArchive, logger) -> None:
+        from nomad.search import search, MetadataPagination
```
Comment (hampusnasstrom) [inline IKZ_plugin/src/ikz_plugin/movpe/movpe1/growth_excel/parser.py:121]: Is there a reason you moved this to inside the parse function? We recently had some issued with these "hidden" imports in the central nomad.

File: IKZ_plugin/src/ikz_plugin/movpe/movpe2/growth_excel/__init__.py [hunk 3]
```diff
@@ -15,5 +15,25 @@
...
 #
 
-from ikz_plugin.movpe.schema import *
-from .parser import *
+from nomad.config.models.plugins import ParserEntryPoint
+from pydantic import Field
+
+
+class Movpe2ParserEntryPoint(ParserEntryPoint):
+    parameter: int = Field(0, description='Custom configuration parameter')
```
Comment (hampusnasstrom) [inline IKZ_plugin/src/ikz_plugin/movpe/movpe2/growth_excel/__init__.py:23]:
```suggestion
```

File: IKZ_plugin/src/ikz_plugin/movpe/movpe2/growth_excel/__init__.py [hunk 4]
```diff
@@ -15,5 +15,25 @@
...
+
+    def load(self):
+        from ikz_plugin.movpe.movpe2.growth_excel.parser import ParserMovpe2IKZ
+
+        return ParserMovpe2IKZ(**self.dict())
+
+
+movpe2_growth_excel_parser = Movpe2ParserEntryPoint(
+    name='Movpe2Parser',
+    description='Parser defined using the new plugin mechanism.',
```
Comment (hampusnasstrom) [inline IKZ_plugin/src/ikz_plugin/movpe/movpe2/growth_excel/__init__.py:33]: Please fill

File: IKZ_plugin/src/ikz_plugin/movpe/movpe2/utils.py [hunk 5]
```diff
@@ -122,6 +120,9 @@ def dict_nan_equal(dict1, dict2):
 def create_archive(
     entry_dict, context, filename, file_type, logger, *, overwrite: bool = False
 ):
+    from nomad.datamodel.context import ClientContext, ServerContext
```
Comment (hampusnasstrom) [inline IKZ_plugin/src/ikz_plugin/movpe/movpe2/utils.py:123]: Again, is there a reason for moving the imports here? You have the same imports again in the next function.

File: IKZ_plugin/src/ikz_plugin/movpe/substrate/__init__.py [hunk 6]
```diff
@@ -15,5 +15,27 @@
...
 #
 
-from ikz_plugin.movpe.schema import *
-from .parser import *
+from nomad.config.models.plugins import ParserEntryPoint
+from pydantic import Field
+
+
+class SubstrateParserEntryPoint(ParserEntryPoint):
+    parameter: int = Field(0, description='Custom configuration parameter')
```
Comment (hampusnasstrom) [inline IKZ_plugin/src/ikz_plugin/movpe/substrate/__init__.py:23]:
```suggestion
```

File: IKZ_plugin/src/ikz_plugin/movpe/substrate/__init__.py [hunk 7]
```diff
@@ -15,5 +15,27 @@
...
+
+    def load(self):
+        from ikz_plugin.movpe.substrate.parser import MovpeSubstrateParser
+
+        return MovpeSubstrateParser(**self.dict())
+
+
+substrate_excel_parser = SubstrateParserEntryPoint(
+    name='SubstrateParser',
+    description='Parser defined using the new plugin mechanism.',
```
Comment (hampusnasstrom) [inline IKZ_plugin/src/ikz_plugin/movpe/substrate/__init__.py:33]: Please fill

File: IKZ_plugin/src/ikz_plugin/pld/__init__.py [hunk 8]
```diff
@@ -1 +1,34 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from nomad.config.models.plugins import SchemaPackageEntryPoint
+from pydantic import Field
+
+
+class PldEntryPoint(SchemaPackageEntryPoint):
+    parameter: int = Field(0, description='Custom configuration parameter')
```
Comment (hampusnasstrom) [inline IKZ_plugin/src/ikz_plugin/pld/__init__.py:23]:
```suggestion
```

File: IKZ_plugin/src/ikz_plugin/pld/__init__.py [hunk 9]
```diff
@@ -1 +1,34 @@
...
+
+    def load(self):
+        from ikz_plugin.pld.schema import m_package
+
+        return m_package
+
+
+pld_schema = PldEntryPoint(
+    name='PldSchema',
+    description='Schema package defined using the new plugin mechanism.',
```
Comment (hampusnasstrom) [inline IKZ_plugin/src/ikz_plugin/pld/__init__.py:33]:
```suggestion
    description='A package containing schemas for Pulsed Laser Deposition (PLD) done at IKZ.',
```

File: IKZ_plugin/src/ikz_plugin/pld/schema.py [hunk 10]
```diff
@@ -29,6 +29,7 @@
     Any,
     TYPE_CHECKING,
 )
+from nomad.config import config
```
Comment (hampusnasstrom) [inline IKZ_plugin/src/ikz_plugin/pld/schema.py:32]:
```suggestion
```

File: IKZ_plugin/src/ikz_plugin/pld/schema.py [hunk 11]
```diff
@@ -112,7 +114,9 @@
         BoundLogger,
     )
 
-m_package = Package(name='IKZ PLD')
+configuration = config.get_plugin_entry_point('ikz_plugin.pld:pld_schema')
```
Comment (hampusnasstrom) [inline IKZ_plugin/src/ikz_plugin/pld/schema.py:117]: I don't use it at least
```suggestion
```

File: Lakeshore_plugin/src/lakeshore_plugin/hall/__init__.py [hunk 12]
```diff
@@ -19,7 +19,7 @@
 from pydantic import Field
 
 
-class SchemaPackageEntryPoint(SchemaPackageEntryPoint):
+class HallEntryPoint(SchemaPackageEntryPoint):
     parameter: int = Field(0, description="Custom configuration parameter")
```
Comment (hampusnasstrom) [inline Lakeshore_plugin/src/lakeshore_plugin/hall/__init__.py:23]:
```suggestion
```

File: Lakeshore_plugin/src/lakeshore_plugin/hall/__init__.py [hunk 13]
```diff
@@ -28,7 +28,7 @@ def load(self):
         return m_package
 
 
-hall_schema = SchemaPackageEntryPoint(
+hall_schema = HallEntryPoint(
     name="HallSchema",
     description="Schema package defined using the new plugin mechanism.",
```
Comment (hampusnasstrom) [inline Lakeshore_plugin/src/lakeshore_plugin/hall/__init__.py:33]:
```suggestion
    description="Schema package for Hall measurement definitions.",
```

**Review by hampusnasstrom — CHANGES_REQUESTED** [review]
Great work! I added some suggestions for descriptions where I knew what to write but there are some parsers where you could maybe add some. Do you use any of the configuration parameters? If not, we should probably remove them. I tried looking though the files but it was a bit hard with all the Ruff linting... If we have some more old code to refactor we could consider merging the Ruff formatting first. I didn't test that it installs and runs all your plugins as I assume you have already tested that.

File: IKZ_plugin/README.md [hunk 14]
```diff
@@ -28,20 +30,66 @@ IKZ_plugin/
...
+to ensure installation of all the packages required, make sure in to install:
+
+```sh
+pip install nomad-lab[parsing, infrastructure]
+```
+
+### Install the plugin without pip
 
-To use these plugins, you need to:
+For some reason, you may want to avoid installation of the package with pip and plug this code to NOMAD in a more direct fashion:
```
[the remark this answers is not in the document]
Comment (aalbino2) [reply IKZ_plugin/README.md:92]: removed

File: IKZ_plugin/README.md [hunk 15]
```diff
@@ -94,82 +142,18 @@ To load the full functionality, use the following `plugins` section:
...
+    - 'ikz_plugin.pld:pld_schema'
+    - 'ikz_plugin.movpe:movpe_schema'
+    - 'ikz_plugin.movpe.movpe2.growth_excel:movpe2_growth_excel_parser'
+    - 'ikz_plugin.movpe.movpe1.growth_excel:movpe1_growth_excel_parser'
+    - 'ikz_plugin.movpe.substrate:substrate_excel_parser'
+    - 'ikz_plugin.mbe:mbe_schema'
+    - 'ikz_plugin.directional_solidification:dir_sol_schema'
+    - 'ikz_plugin.directional_solidification:dir_sol_manual_protocol_excel_parser'
+    - 'ikz_plugin.czochralski:czochralski_schema'
+    - 'ikz_plugin.czochralski:czochralski_multilog_parser'
```
[the remark this answers is not in the document]
Comment (aalbino2) [reply IKZ_plugin/README.md:156]: yeah, I dropped this section..

File: IKZ_plugin/src/ikz_plugin/characterization/schema.py [hunk 16]
```diff
@@ -3,15 +3,21 @@
...
+
+from ikz_plugin.general.schema import (
     IKZCategory,
     SubstratePreparationStep,
 )
 
-m_package = Package(name="characterization_IKZ")
+configuration = config.get_plugin_entry_point(
+    'ikz_plugin.characterization:characterization_schema'
+)
```
[the remark this answers is not in the document]
Comment (aalbino2) [reply IKZ_plugin/src/ikz_plugin/characterization/schema.py:18]: I will leave it here, even if not used now

**Pull request merged**